code-challenges.test.js

```diff
@@ -14,47 +16,116 @@
 
+// make array length be the number of input. I did not need to do this but i originally thought i would. i just set the loop to num.
+//number 1 and number 2 must = number 3
+//take 3rd number and make it equal to itself minus 2 and minus 1 to get number 3.
+//loop thru and make sure to go length of num
+
+function fibonacci(num) {
+    let array = [];
+    array[0] = 1;
+    array[1] = 1;
+    for(i = 2; i <num; i++){
+        array[i] = array[i - 2] + array[i - 1]
+    }
+    return array;
+}
 
 
 // --------------------2) Create a function that takes in an array and returns a new array of only odd numbers sorted from least to greatest.
 
 // a) Create a test with expect statements for each of the variables provided.
+//test should expect to equal expected output 
+
+describe('oddSorter', () => {
+    it('returns an array with the odds sorted from least to greatest', () => {
+        expect(oddSorter(fullArr1)).toEqual([-9, 7, 9, 199])
+        expect(oddSorter(fullArr2)).toEqual([-823, 7, 23])
+    })
+})
 
 var fullArr1 = [4, 9, 0, "7", 8, true, "hey", 7, 199, -9, false, "hola"]
 // Expected output: [-9, 7, 9, 199]
 
 var fullArr2 = ["hello", 7, 23, -823, false, 78, null, "67", 6, "number"]
 // Expected output: [-823, 7, 23]
 
+//create a blank array
+//iterate thru the array and take anything out not a number and not odd. 
+//Do Math absolute for negative numbers
+//use sorting method
 
-// b) Create the function that makes the test pass.
+function oddSorter(array){
+    let final = [];
+    for(let i=0; i<array.length; i++){
+        if(typeof array[i] == 'number' && array[i] % 2 === 1){
+            final.push(array[i])
+         } 
+         if(array[i] < 0){
+             Math.abs(array[i])
+             final.push(array[i])
+         }
+        }
+    return final.sort((a, b) => a - b);
+}
+
+oddSorter(fullArr1);
+oddSorter(fullArr2);
 
 
 
+// b) Create the function that makes the test pass.
+
+
 // --------------------3) Create a function that takes in an array and returns an array of the accumulating sum. An empty array should return an empty array.
 
 // a) Create a test with expect statements for each of the variables provided.
+//test should have expected of expected output for each scenario
 
 var numbersToAdd1 = [2, 4, 45, 9]
-// Excpected output: [2, 6, 51, 60]
+// Expected output: [2, 6, 51, 60]
 
 var numbersToAdd2 = [0, 7, -8, 12]
 // Expected output: [0, 7, -1, 11]
 
 var numbersToAdd3 = []
 // Expected output: []
 
-
+describe('cumSum', () => {
+    it('returns an array with the accumulated sum', () => {
+        expect(cumSum(numbersToAdd1)).toEqual([2, 6, 51, 60])
+        expect(cumSum(numbersToAdd2)).toEqual([0, 7, -1, 11])
+        expect(cumSum(numbersToAdd3)).toEqual([])
+        
+    })
+})
 
 // b) Create the function that makes the test pass.
+//blank arrays return blank arrays
+//number 1 + number 2 = number i
+
+let cumSum = array => array.map((sum => value => sum += value)(0));    
```

Comment:
Alright that makes a lot more sense when I don't have 5 ways I am trying to solve it. Still when we (anyone: junior mid or senior) are bringing in dense syntax we should take a moment to leave a note of how this syntax is work and why it is working. Line 122 does make sense in this regard but expand on it a bit more and talk about why its a better methodology.

Reply:
Thanks for the feedback Austin. Especially if we talking about getting me some more money!! haha appreciate it